test/integration/dnsrecord/dnsrecord_test.go

```diff
@@ -0,0 +1,294 @@
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+var (
+	accessKeyID     = flag.String("access-key-id", "", "AWS access key id")
+	secretAccessKey = flag.String("secret-access-key", "", "AWS secret access key")
+	zoneName        = flag.String("zone-name", "", "AWS Route53 zone name")
+	zoneID          = flag.String("zone-id", "", "AWS Route53 zone ID")
+)
+
+func validateFlags() {
+	if len(*accessKeyID) == 0 {
+		panic("need an AWS access key id")
+	}
+	if len(*secretAccessKey) == 0 {
+		panic("need an AWS secret access key")
+	}
+	if len(*zoneName) == 0 {
+		panic("need an AWS Route53 zone name")
+	}
+	if len(*zoneID) == 0 {
+		panic("need an AWS Route53 zone ID")
+	}
+}
+
+var _ = Describe("DNSRecord tests", func() {
+	var (
+		ctx = context.Background()
+
+		logger    *logrus.Entry
+		awsClient *awsclient.Client
+		testEnv   *envtest.Environment
+		mgrCancel context.CancelFunc
+		c         client.Client
+
+		testName  = fmt.Sprintf("aws-dnsrecord-it--%s", randomString())
+		namespace = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: testName,
+			},
+		}
+		secret = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "dnsrecord",
+				Namespace: testName,
+			},
+			Data: map[string][]byte{
+				aws.AccessKeyID:     []byte(*accessKeyID),
+				aws.SecretAccessKey: []byte(*secretAccessKey),
+			},
+		}
+		cluster = &extensionsv1alpha1.Cluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: testName,
+			},
+		}
+	)
+
+	BeforeSuite(func() {
+		repoRoot := filepath.Join("..", "..", "..")
+
+		// enable manager logs
+		logf.SetLogger(zap.New(zap.UseDevMode(true), zap.WriteTo(GinkgoWriter)))
+
+		log := logrus.New()
+		log.SetOutput(GinkgoWriter)
+		logger = logrus.NewEntry(log)
+
+		By("starting test environment")
+		testEnv = &envtest.Environment{
+			UseExistingCluster: pointer.BoolPtr(true),
+			CRDInstallOptions: envtest.CRDInstallOptions{
+				Paths: []string{
+					filepath.Join(repoRoot, "example", "20-crd-dnsrecord.yaml"),
+					filepath.Join(repoRoot, "example", "20-crd-cluster.yaml"),
+				},
+			},
+		}
+
+		cfg, err := testEnv.Start()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(cfg).ToNot(BeNil())
+
+		By("setting up manager")
+		mgr, err := manager.New(cfg, manager.Options{
+			MetricsBindAddress: "0",
+		})
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(extensionsv1alpha1.AddToScheme(mgr.GetScheme())).To(Succeed())
+		Expect(awsinstall.AddToScheme(mgr.GetScheme())).To(Succeed())
+
+		Expect(dnsrecordctrl.AddToManager(mgr)).To(Succeed())
+
+		var mgrContext context.Context
+		mgrContext, mgrCancel = context.WithCancel(ctx)
+
+		By("starting manager")
+		go func() {
```

Comment:
See doc string of `GinkgoRecover`

```suggestion
		go func() {
		     defer GinkgoRecover()
```

Reply:
Thanks, will take a look.

Reply:
Done